api/scpca_portal/metadata_file.py

```diff
@@ -87,6 +88,49 @@ def transform_keys(data_dict: Dict, key_transforms: List[Tuple]):
     return data_dict
 
 
+class MetadataFilenames:
+    SINGLE_CELL_METADATA_FILE_NAME = "single_cell_metadata.tsv"
+    SPATIAL_METADATA_FILE_NAME = "spatial_metadata.tsv"
+    METADATA_ONLY_FILE_NAME = "metadata.tsv"
+
+
+def get_file_name(download_config: Dict) -> str:
+    """Return metadata file name according to passed download_config."""
+    if download_config.get("metadata_only", False):
+        return MetadataFilenames.METADATA_ONLY_FILE_NAME
+
+    return getattr(MetadataFilenames, f'{download_config["modality"]}_METADATA_FILE_NAME')
+
+
+def get_file_contents(libraries, **kwargs) -> str:
+    """Return newly genereated metadata file as a string for immediate writing to a zip archive."""
+    libraries_metadata = [
+        format_metadata_dict(library_metadata)
+        for library in libraries
+        for library_metadata in library.get_combined_library_metadata()
+    ]
```

Comment:
Making this change would require an overhaul of the tests so let's keep this as the callers responsibility for now. You should open an issue that covers this change and the required changes to testing there.